src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -283,6 +283,24 @@ InvokeHostFunctionOpFrame::doApply(AbstractLedgerTxn& ltx, Config const& cfg,
         innerResult().code(INVOKE_HOST_FUNCTION_TRAPPED);
         return false;
     }
+    
+    // Append events to the enclosing TransactionFrame, where
+    // they'll be picked up and transferred to the TxMeta.
+    for (auto const& buf : out.contract_events)
```

Comment:
I don't think we want to report events on error -- the transaction is going to be rolled back, which implies that whatever the events are describing didn't actually happen.

Reply:
Sounds good, I confused those with the debug events which could be meaningful for errors. Thanks for fixing this.